### PR TITLE
Fix reconnection of client connection

### PR DESCRIPTION
**Pull request by damencho**

### CODE REVIEW

File: src/main/java/org/jitsi/impl/protocol/xmpp/XmppProtocolProvider.java [hunk 1]
```diff
@@ -288,6 +295,12 @@ synchronized private boolean doConnect()
             // rely on Smack's built-in retries, as it will be handled by
             // the RetryStrategy
             connection.removeConnectionListener(connListener);
+
+            ReconnectionManager reconnectionManager =
```
Comment (paweldomas) [inline src/main/java/org/jitsi/impl/protocol/xmpp/XmppProtocolProvider.java]: shouldn't the equals sign be moved to the next line ?

Comment (damencho) [reply src/main/java/org/jitsi/impl/protocol/xmpp/XmppProtocolProvider.java]: ups

File: src/main/java/org/jitsi/impl/protocol/xmpp/XmppProtocolProvider.java [hunk 2]
```diff
@@ -248,6 +255,12 @@ synchronized private boolean doConnect()
             connection.connect();
 
             connection.addConnectionListener(connListener);
+            ReconnectionManager.getInstanceFor(connection)
+                .addReconnectionListener(reConnListener);
+
+            discoInfoManager = new ScServiceDiscoveryManager(
```
Comment (paweldomas) [inline src/main/java/org/jitsi/impl/protocol/xmpp/XmppProtocolProvider.java]:
```
discoInfoManager
    = new ScServiceDiscoveryManager(
            XmppProtocolProvider.this,
            connection,
            new String[]{},
            new String[]{},
            false);
```

File: src/main/java/org/jitsi/impl/protocol/xmpp/XmppProtocolProvider.java [hunk 3]
```diff
@@ -248,6 +255,12 @@ synchronized private boolean doConnect()
             connection.connect();
 
             connection.addConnectionListener(connListener);
+            ReconnectionManager.getInstanceFor(connection)
```
Comment (paweldomas) [inline src/main/java/org/jitsi/impl/protocol/xmpp/XmppProtocolProvider.java]:
```
ReconnectionManager
        .getInstanceFor(connection)
        .addReconnectionListener(reConnListener);
```

**Comment by damencho** [comment]
Just fixed them.

**Pull request merged**